src/lib/pubkey/sphincsplus/sphincsplus_common/sphincsplus.cpp

```diff
@@ -19,14 +20,40 @@
 #include <botan/internal/sp_xmss.h>
 #include <botan/internal/stl_util.h>
 
-#if !defined(BOTAN_HAS_SPHINCS_PLUS_WITH_SHA2) and !defined(BOTAN_HAS_SPHINCS_PLUS_WITH_SHAKE)
+#include <utility>
+
+#if !defined(BOTAN_HAS_SPHINCS_PLUS_WITH_SHA2) and !defined(BOTAN_HAS_SPHINCS_PLUS_WITH_SHAKE) and \
+   !defined(BOTAN_HAS_SLH_DSA_WITH_SHA2) and !defined(BOTAN_HAS_SLH_DSA_WITH_SHAKE)
 static_assert(
    false,
-   "botan module 'sphincsplus_common' is useful only when enabling at least 'sphincsplus_sha2' or 'sphincsplus_shake'");
+   "botan module 'sphincsplus_common' is useful only when enabling at least 'sphincsplus_sha2', 'sphincsplus_shake', 'slh_dsa_sha2', or 'slh_dsa_shake'");
 #endif
 
 namespace Botan {
 
+namespace {
+// FIPS 205, Algorithm 22, line 8
+SphincsMessageInternal prepare_message(SphincsInputMessage&& msg,
```

Comment:
Do you have a good reason to enforce that `SphincsInputMessage` is an rvalue. (I see that it is coming from a `std::exchange` but that doesn't demand the rvalue-ref here, or does it?)

```suggestion
SphincsMessageInternal prepare_message(SphincsInputMessage msg,
```

Reply:
Since the input message has been moved into the SphincsMessageInternal struct, I need an rvalue or a copy. I want to prevent us from accidentally copying the full message, so I used an rvalue-ref. Do you think this is problematic?

Reply:
There's nothing (technically) wrong with it. Especially not for such an internal interface.

This is really more of a personal style preference.

Reply:
I'll change it then, since both versions are fine to me :)